src/errors.rs

```diff
@@ -0,0 +1,17 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+#[error("rlimit error: {0}")]
+pub struct RlimitsError(pub(crate) String);
```

Comment:
`ParseResourceError` is better, for example, <https://doc.rust-lang.org/std/index.html?search=Parse>.

A parse error is not a string, it usually likes this: <https://doc.rust-lang.org/src/core/str/mod.rs.html#149-151>.

There is usually only one way to construct a parse error: `from_str`. We should not provide `From` implementations.